### DISCUSSION TITLE
Gravity processing, Equivalent Sources and Levelling

### DISCUSSION THREAD

**Discussion by mdtanker** [fatiando/community#70] [Q&A] [answered]
I was wondering if anyone could explain a bit about how and where the equivalent sources technique fits into the workflow for processing gravity data (specifically airborne).  For airborne surveys I believe the standard processing workflow is: 

* 1: correct the raw data for machine drift, tides, aircraft manoeuvres and the Eötvös correction.
* 2: level the flightlines and tie-lines to minimize crossovers errors. This should give you **Observed Gravity**.
* 3: Calculate the **free-air anomaly / gravity disturbance** (these terms are used interchangeably in the literature correct?):
     * ideally use a closed-form normal gravity formula (such as in Boule) to correct for both latitude and observation height at the same time, OR
     * use the standard latitude correction and free-air correction to get a free-air anomaly

Is this where you would implement Equivalent Sources to upward continue the FAA/gravity disturbance to a consistant height? 

The troubling part for me is what occurs at flight line crossover points. As in the figure below, at cross-overs there will be multiple adjacent FAA values, which should (might not) have similar values due to the flight line levelling, but, they will likely have different observation elevations (tie lines vs flight lines may have been at different altitude). A block-reduction (middle image) can give the median altitude and gravity values, but I'm not sure if this is the appropriate solution. The right figure shows a prediction of the gravity disturbance at 1km altitude, using Eq sources. 

Does anyone here have experience with is? My initial guess is to just pick either the flight lines or the tielines to use at the crossovers, instead of doing a block median. 

ROSETTA-Ice airborne gravity data | 1km block-median | equivalent source solution at 1km altitude
:---: | :---: | :---:
![image](https://user-images.githubusercontent.com/81199856/198161992-ed52f3e6-5419-4d99-9150-2ef3d2e4a5c0.png) | ![image](https://user-images.githubusercontent.com/81199856/198162044-b64b7eba-391b-4ccd-a4fa-cfb56ab7c1fb.png) | ![image](https://user-images.githubusercontent.com/81199856/198162319-cf14af16-effd-41bb-919a-757af3d2c5a5.png)

(Note; the lat lon gridline annotations should have decimal places)

Relevant links:
[Free-air correction vs normal gravity](https://github.com/fatiando/harmonica/issues/164)
[Harmonica issue on airborne corrections](https://github.com/fatiando/harmonica/issues/235)
[Example airborne report](http://www.geosciencebc.com/i/project_data/QUESTdata/GBCReport2008-8/Gravity_Technical_Report.pdf)

**Comment by santisoler** [comment, marked as answer]
Hi @mdtanker! Thanks for opening this discussion.

I think a great part of what you are asking is related to what do you want to do with the data in the end. So, I would start with the same first point:

> 1: correct the raw data for machine drift, tides, aircraft manoeuvres and the Eötvös correction.

After this you would have a set of observation points that you can call the **observed gravity**. This means that the signal in there is produced by the gravitational attraction of every massive body in the Earth plus an extra acceleration term produced by the fictitious centrifugal force that appears because we are not measuring the acceleration from an inertial reference system. The **observed gravity** is defined on observation points that follow flight lines at different heights, and some of them "crossover" in the longitude-latitude projected plane, but they are not actually crossing each other because of the difference in height. This means that you have some observations in approximately the same longitude and latitude coordinates, but at different heights. And that's ok. In fact, for running inversions, this is a plus: you have some degree of information about how the gravity acceleration changes in the vertical direction.

My next step would be to compute the **gravity disturbance** by removing the _normal gravity_ (the gravity acceleration produced by the reference ellipsoid). The normal gravity is equal to the gravitational acceleration of the ellipsoid + the centrifugal term. So the gravity disturbance holds the signal that corresponds to the gravitational attraction of every _anomalous body_ in the Earth. The term free-air anomaly is sometimes used interchangeably with the gravity disturbance, but they are actually different magnitudes. Take a look at this: https://www.pinga-lab.org/pdf/use-the-disturbance.pdf. The main differences are:

1. The gravity anomaly vector is defined as the observed gravity acceleration vector on a point located on the geoid minus the normal gravity vector at a point located in the ellipsoid (where the two points hold the same longitudinal and latitudinal coordinates). The _gravity anomaly_ is the difference between the norm of the observed gravity acceleration vector and the norm of the normal gravity vector.
While the gravity disturbance vector is the difference between the observed gravity acceleration vector and the normal gravity vector at the same location. The gravity disturbance is the difference between the norm of the observed gravity acceleration vector and the norm of the normal gravity vector.
2. The second difference relies on how the normal gravity is considered. Since in the gravity anomaly the normal gravity is computed on the ellipsoid, the free-air correction is introduced to account for the variation of the normal gravity with height. But in the gravity disturbance we are removing the normal gravity through an analytical solution. So basically, the free-air anomaly performs a first (or second) degree taylor expansion of the normal gravity, while with the gravity disturbance we solve it through the analytical solution.

So, by the end of this step you would have values of the **gravity disturbance** on every observation point (even in those flight lines at different heights). Don't apply block reductions here, since the data that fall under a single block have noticeably different heights, and therefore the data values in there cannot be averaged.

Now, if you want to have a map that could help you visualizing the data for interpretation I would use equivalent sources to predict the gravity disturbance values on a regular grid at a constant height. This height could be the maximum flight height if you want to make sure that you are not downcontinuing the data. But if you are going to run an inversion, I would include the data as it is, with the flight lines at different height.

A third step I would add is a terrain correction. The gravity disturbance is mostly governed by the effect of the topographic masses because they are the closest ones to the observation points and because their density contrast is very high (density of the rocks - density of air ~ density of the rock). That would leave you with the Bouger gravity disturbance, which accounts for all the anomalous masses in the Earth except for the topographic ones that you removed.

I hope this is helpful Matt! Happy to answer more questions!

**Reply by mdtanker** [reply]
Hi Santi, that is an excellent summary, thanks so much for taking the time to write it up! Reading through your explanation, and the linked paper has really cleared up some misconceptions for me. 

For the _gravity disturbance_, it is always calculated on the ellipsoid for the observation points since we can only calculate the normal gravity on the ellipsoid as well, correct? Theoretically, if we could calculate normal gravity on the geoid, we could use the observation points on the geoid?

Regarding flight line leveling, the cross-over errors are partially due to flight height differences, and partially due to inaccuracies in the machine drift correction, since we can't revisit base stations frequently enough during long flights. Do you think it's best to leave all leveling to be done after the gravity disturbance calculation since this will change the cross-over errors? 

Does anyone here know of any open-source tools for flight line leveling? I'm familiar with the leveling in Oasis Montaj, but it seems like a similar tool wouldn't be too hard to include in Harmonica. I'll add a link to this discussion to @leouieda's [Harmonica issue for common airborne corrections](https://github.com/fatiando/harmonica/issues/235).

**Reply by leouieda** [reply]
> since we can only calculate the normal gravity on the ellipsoid as well, correct?

With appropriate equations like what we have in Boule, you can calculate normal gravity anywhere outside of the ellipsoid (without a free-air correction). The main difference is that for a gravity *anomaly*, the goal of the correction is to get the *observed* gravity to the geoid since that's the actual definition of a gravity anomaly. And that only makes sense in geodesy, where the goal is to calculate the geoid. In geophysics, we're not interested in that and the only thing that makes any sense is the gravity *disturbance*. In the disturbance, the observed gravity is what it is and we calculate normal gravity at the same point (hence why we use ellipsoid heights).  

> Regarding flight line leveling, the cross-over errors are partially due to flight height differences, and partially due to inaccuracies in the machine drift correction

This is at the heart of why I never understood cross-over analysis for potential fields. When measuring topography or altimetry, it makes sense because the observation doesn't depend on the instrument position. But for gravity and magnetics there are two effects at play and it's hard to tease them apart. 

Do you know how Oasis does this? From your figures, the cross-over points don't seem to have the same gravity value. So was the height taken into account (using a free-air gradient to approximate the difference due to heights)?

With equivalent sources, if you try to fit the data before leveling, ideally the model shouldn't be able to fit both the flight and tie lines if there is drift still in the data. So maybe that could be a way to figure out the drift correction? Or build an equivalent source model that includes a linear drift term?

> Does anyone here know of any open-source tools for flight line leveling?

GMT has this capability: https://www.researchgate.net/publication/220164039_Tools_for_analyzing_intersecting_tracks_The_x2sys_package I don't think it's in PyGMT yet since this is a tricky add-on to make Pythonic.

**Reply by leouieda** [reply]
Also, this is an excellent book on the subject of gravity anomalies and disturbances: https://www.sciencedirect.com/book/9780128129135/understanding-the-bouguer-anomaly

**Reply by mdtanker** [reply]
> This is at the heart of why I never understood cross-over analysis for potential fields. 

Could a solution be to _separately_ upward continue (with eq sources) the tie lines and flight lines to a constant elevation in order to calculate accurate mis-ties, and then use those mis-ties in the raw data to level the lines? Hopefully, this would isolate the portion of the mis-ties just related to errors in machine drift? 

> Do you know how Oasis does this?

From what I've seen, Oasis just gives you recommendations for how to level the data, but not at which stage to perform the levelling. Their recommended process is to statistically level (recommend a shift or tilt, but you can use a spline or tensioned level) the tie-lines to the flight lines, recalculate the mis-ties, then perfectly match the flight-lines to the newly leveled tie lines, and linearly interpolate the difference between adjacent cross-over points. 

> From your figures, the cross-over points don't seem to have the same gravity value. 

This is a small portion of a very large survey, so I think this is the best leveling they were able to achieve without using high-order polynomials for levelling. That N-S tieline has about 50 total cross-overs with flightlines.

> So was the height taken into account (using a free-air gradient to approximate the difference due to heights)?

No, I don't think it was, but I guess on average for all the survey's cross-overs the flight elevations were similar so maybe it was an alright approx for such a large survey. 

Thanks for pointing out that x2sys package. It looks like x2sys_init and x2sys_cross are wrapped by pygmt, but not yet x2sys_solve, which is what would be needed to apply the mis-tie corrections. It also seems to be a bit of a cumbersome software to use. If I end up having to re-level this dataset I'll try x2sys, and if it's not working well I'll look into coding something myself.

**Reply by mdtanker** [reply]
@santisoler, could you please expand a little on the below:

> ... observed gravity. This means that the signal in there is produced by the gravitational attraction of every massive body in the Earth plus an extra acceleration term produced by the fictitious centrifugal force that appears because we are not measuring the acceleration from an inertial reference system.

I thought that the Eotvos correction (I thought which had already been removed from the observed gravity) was accounting for that extra acceleration? What makes it fictitious?

**Reply by santisoler** [reply]
I'm not an expert on the Eotvos correction, but from what I understand it's accounting for how the centrifugal acceleration changes when you're meassuring the gravity on a moving object with respect to a meassure on a fixed point on the Earth surface. Since you are moving with respect to Earth surface, your centrifugal acceleration cannot be computed using the Earth's angular velocity, since your angular velocity is different (you are actually moving a different angular velocity than the Earth). So the Eotvos correction accounts for that difference, but it doesn't remove the centrifugal acceleration from the observed gravity.

The centrifugal force it's called fictitious because it's not being generated by any interaction with any other particle. It appears only because we are standing on a non inertial frame of reference (where Newton laws are not satisfied). The centrifugal and the Coriolis forces are examples of these: https://en.wikipedia.org/wiki/Fictitious_force

In this sentence:
> This means that the signal in there is produced by the gravitational attraction of every massive body in the Earth plus an extra acceleration term produced by the fictitious centrifugal force that appears because we are not measuring the acceleration from an inertial reference system.

I was trying to put in words that:

$$ g_\text{obs}(\mathbf{p}) = \delta g(\mathbf{p}) + \gamma(\mathbf{p}) $$

The observed gravity  is equal to the gravity disturbance plus the normal gravity (all on the same observation point p). Inside the normal gravity we have two terms: the gravitational attraction of the reference ellipsoid and the centrifugal acceleration produced by the fact that the observation point p is rotating with the Earth.

**Comment by mdtanker** [comment]
Are there any issues with _undoing_ the Free-air and latitude corrections of a FAA, in order to go back to the observed gravity, so I can properly calculate the gravity disturbance, according to Santi's outlined method above?

**Reply by santisoler** [reply]
I don't see any problem. The classic free-air correction is a subtraction that can be easily undone. The only issue is knowing exactly how it was done (first or second order free-air correction? which values for the parameters do they use?)

**Reply by leouieda** [reply]
Exactly. It's often the case that older surveys just preserve free air and Bouguer and it's impossible to find out how they were calculated. But if you know, then you could undo them. Most normal gravity corrections use pretty straight forward equations (nothing like what we have in Boule).
